File: 442-find-all-duplicates-in-an-array/442-find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        seen = {}
        for x in nums:
            if seen.get(x,-1) == -1:
                seen[x]=1
            else:
                seen[x]+=1
        res = []
        for x in nums:
            if seen[x] != 1:
                res.append(x)
                seen[x] = 1
        return res

File: 442-find-all-duplicates-in-an-array/442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> res;
        unordered_map<int,int> m;
        
        for(auto a: nums)m[a]++;
        for(auto a:nums){
            if(m[a]==2){
                 res.push_back(a);
                m[a] = -1;
            }
        }
        return res;
    }
};